### ISSUE TITLE
What is the process to install guvnor correctly on ubuntu

### ISSUE THREAD

**Issue by kevincaradant** [tableflip/guvnor#89]
Hi

Sorry , i'm a little bit lost. I tried to start your monitoring-web but i don't understand where are the files that i have to configure. 

i get always this : 
`error: Guvnor has been configured to run as root:guvnor but the guvnor group does not exist.
error: 
error: To fix this run guvnor as root and it will attempt to create the guvnor group.
error: 
error: guvnor group does not exist
`

1) i cloned this repo
2) i write : `npm install -g  --unsafe-perm`
3) guv-web useradd alex

after that , i haven't any config files :/ , is it normal , 2 hours to try your app but i failed every time :(.
What i'm suppose to do ? 

Thank you for your help :)

**Comment by achingbrain** [comment]
You don't need to clone the repo, just install it from npm with:

``` sh
$ npm install -g --unsafe-perm guvnor
```

Start guvnor as root and it will walk you through creating the `guvnor` group:

``` sh
$ sudo guvnor
```

After that you can add a user to the web interface with:

``` sh
$ guv-web useradd kevin
```

Then start the web interface:

``` sh
$ guv web
```

..and open [https://localhost:3001](https://localhost:3001)

**Comment by kevincaradant** [comment]
thank you :) , i get this error : 

```
> mdns@2.2.11 install /home/kevin/npm/lib/node_modules/guvnor/node_modules/mdns
> node-gyp rebuild
make: Entering directory '/home/kevin/npm/lib/node_modules/guvnor/node_modules/mdns/build'
  CXX(target) Release/obj.target/dns_sd_bindings/src/dns_sd.o
In file included from ../src/dns_sd.cpp:1:0:
../src/mdns.hpp:32:20: fatal error: dns_sd.h: Aucun fichier ou dossier de ce type
compilation terminated.
dns_sd_bindings.target.mk:144: recipe for target 'Release/obj.target/dns_sd_bindings/src/dns_sd.o' failed
make: *** [Release/obj.target/dns_sd_bindings/src/dns_sd.o] Error 1
make: Leaving directory '/home/kevin/npm/lib/node_modules/guvnor/node_modules/mdns/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Linux 4.2.0-23-generic
gyp ERR! command "/usr/bin/nodejs" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/kevin/npm/lib/node_modules/guvnor/node_modules/mdns
gyp ERR! node -v v5.3.0
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok 
npm WARN install:mdns@2.2.11 mdns@2.2.11 install: `node-gyp rebuild`
npm WARN install:mdns@2.2.11 Exit status 1
> guvnor@3.5.17 postinstall /home/kevin/npm/lib/node_modules/guvnor
> chmod +x scripts/init/sysv/guvnor
```

Edit : apt-get install libavahi-compat-libdnssd-dev resolved my problem about his error of compilation

My problem now : 
i try after that `sudo guvnor` , and i get : `sudo: guvnor : command not found`

Log :

```
kevin@kevin-Lenovo-Flex-2-14:~/guvnor$ sudo guvnor   
sudo: guvnor : commande introuvable   
kevin@kevin-Lenovo-Flex-2-14:~/guvnor$ guvnor   
error: Guvnor has been configured to run as root:guvnor but the guvnor group does not exist.   
error:    
error: To fix this run guvnor as root and it will attempt to create the guvnor group.    
error:    
error: guvnor group does not exist   
kevin@kevin-Lenovo-Flex-2-14:~$ sudo su   
root@kevin-Lenovo-Flex-2-14:/home/kevin# guvnor   
guvnor : commande introuvable   
root@kevin-Lenovo-Flex-2-14:/home/kevin#    
```

i already try your command after :  
this is the result : 

```
kevin@kevin-Lenovo-Flex-2-14:~$ guv-web useradd kevin
A user with the name kevin already exists
kevin@kevin-Lenovo-Flex-2-14:~$ guv web
error: Guvnor has been configured to run as root:guvnor but the guvnor group does not exist.
error: 
error: To fix this run guvnor as root and it will attempt to create the guvnor group.
error: 
error: guvnor group does not exist
```

Edit 2 : ok after multiples try in root , i write guvnor , it worked :) now when i write `guv web` : 

error: guvnor-web failed to start

any idea ? :)

Edit 3 : it's really weird but in fact i tried that : `$CONFIG_DIR/guvnor-web-hosts` but `$CONFIG_DIR` is not defined on my computer so when i want to start guv-web , that said me : `It should be at /root/.config/guvnor/guvnor-web-hosts` , so i created at the hand , this file with the config from `sudo guv remoteconfig` and finally `guv web` started :).  this method is complicated no ? , it's just my case maybe  ?

**Comment by kevincaradant** [comment]
Sorry for all this message but now i get this error during the try to install .git of your app from the web site : 

my_project on kevin-Lenovo-Flex-2-14 has failed to install - Not authorised , why  ?

**Comment by kevincaradant** [comment]
ok now i tried to all delete, in no root , when i wrote `guvnor` command , i get : `error: The daemon is not running. Please run this command as root to start the daemon.` so i try with root, i get this `guvnor : command not found` , so i decide to write `npm install -g --unsafe-perm guvnor` and after that the command `guvnor` give me this : No process Running , so i have to install guvnor with sudo right ? 

Seriously what's wrong with my install ? problem with npm or something like that ? 

After alwways like before i get this : 
root@kevin-Lenovo-Flex-2-14:/home/kevin# guv web
`error: guvnor-web failed to start`

So i edited the file in /root/.config/guvnor/guvnor-hosts , where it wanted the conf So after that,  guvnor starts :) but every time i want to add an app for example With the clone
 Of your repository,  guvnor says,  not authorized With à Red popup :/

**Comment by achingbrain** [comment]
Are you using a node version manager like nvm or n or similar?

**Comment by kevincaradant** [comment]
Il dont think so. I dont know nvm but i used nodejs 5.4,  npm 3.5 and i put npm under /home/kevin/npm/lib in the goal to use npm without sudo rights,  that all.  

Also if i do that : `guv install <url>` in sudo that work and i see the app in web site but otherwise i can't add this git directly from the web site :/ . Have i to run the web site with a particular way ?

**Comment by kevincaradant** [comment]
Also how can have I the good rights with this error ? : 

```
npm ERR! Linux 4.2.0-16-generic
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "install" "-g" "--unsafe-perm" "guvnor"
npm ERR! node v4.2.4
npm ERR! npm  v2.14.12
npm ERR! path ../lib/node_modules/guvnor/bin/guv
npm ERR! code EACCES
npm ERR! errno -13
npm ERR! syscall symlink

npm ERR! Error: EACCES: permission denied, symlink '../lib/node_modules/guvnor/bin/guv' -> '/usr/bin/bs'
npm ERR!     at Error (native)
npm ERR!  { [Error: EACCES: permission denied, symlink '../lib/node_modules/guvnor/bin/guv' -> '/usr/bin/bs']
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'symlink',
npm ERR!   path: '../lib/node_modules/guvnor/bin/guv',
npm ERR!   dest: '/usr/bin/bs' }
npm ERR! 
npm ERR! Please try running this command again as root/Administrator.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/monpc/npm-debug.log
```

i  have already do : 
sudo chown -R $USER /usr/local
sudo chown -R $USER /usr/lib/node_modules

so i don't know what can i do if i want to instal it without "sudo" which fix the problem but it's not good i think

**Comment by kevincaradant** [comment]
I want so much use your work , two daysto try to install and i can't have a stable version without problem on guv-web or guvnor-web process which crashed, i tried on 3 different computers but i have always different problems with rights access or bad config file etc , how is possible , it's only me ? If i follow step by step your commands that you mentioned before here , that does not work. i'm ready to follow and configure as you want my install to try but , i don't know what can i do now :(

Edit : To install nodejs and also npm by the same way , i execute on the computer : 

```
curl -sL https://deb.nodesource.com/setup_4.x | sudo -E bash -
sudo apt-get install -y nodejs
```

i have a ubuntu 15.10 64 bits 

is it good ?

**Comment by danyim** [comment]
I'm running into the same issues as @kevincaradant. I am on Xubuntu 13.03 using NVM to manage Node 5.5.0.

At this point, I am going to try the Docker route instead.

**Comment by kevincaradant** [comment]
ok , don't hesitate to give your solution if you find it. thank you :)

**Comment by danyim** [comment]
Docker didn't work for me because my home server is x86. I tried the npm route again, but there's a bug with Node where certain dependencies for certain architectures attempt to be installed when they're incompatible. [Here's the issue for reference.](https://github.com/npm/npm/issues/10768#issuecomment-182173658)

Trying a clean slate with an old Node/npm version, I ran the following

```
nvm install 5.2.0
npm cache clean -f
npm install -g guvnor --unsafe-perm
```

...and that worked for me.
